redbot/core/bank.py

```diff
@@ -345,6 +351,11 @@ async def set_balance(member: Union[discord.Member, discord.User], amount: int)
     if await group.name() == "":
         await group.name.set(member.display_name)
 
+    payload = BankSetBalanceInformation(
+        member.id, (0 if _cache_is_global else guild.id), old_balance, amount
```

Comment:
It's more typical for `guild` like attributes to be `Optional` and use `None` as the value to represent non-guild actions (IE, `message.guild`). If it is consistently possible to provide the full user/guild object rather than an ID, I also think that would be preferred.